src/uct/ib/base/ib_iface.c

```diff
@@ -479,6 +479,40 @@ static ucs_status_t uct_ib_iface_create_cq(uct_ib_iface_t *iface, int cq_length,
     return status;
 }
 
+
+static ucs_status_t uct_ib_iface_set_moderation(struct ibv_cq *cq,
+                                                const uct_ib_iface_config_t *config)
+{
+    ucs_status_t status = UCS_OK;
+
+#if HAVE_DECL_IBV_EXP_CQ_MODERATION
+    if (config->cq_moderation_count > UINT16_MAX) {
+        ucs_error("CQ moderation count is too high: %u", config->cq_moderation_count);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if (config->cq_moderation_period > UINT16_MAX) {
+        ucs_error("CQ moderation period is too high: %u", config->cq_moderation_period);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if (config->cq_moderation_count < UINT16_MAX ||
+        config->cq_moderation_period < UINT16_MAX) {
+        struct ibv_exp_cq_attr cq_attr = {
+            .comp_mask            = IBV_EXP_CQ_ATTR_MODERATION,
+            .moderation.cq_count  = (uint16_t)(config->cq_moderation_count),
+            .moderation.cq_period = (uint16_t)(config->cq_moderation_period)
+        };
+        if (ibv_exp_modify_cq(cq, &cq_attr, IBV_EXP_CQ_MODERATION)) {
+            ucs_error("ibv_exp_modify_cq() failed: %m");
+            status = UCS_ERR_IO_ERROR;
+        }
+    }
+#endif /* HAVE_DECL_IBV_EXP_CQ_MODERATION */
+
+    return status;
```

Comment:
you could just return UCS_OK here, status is not needed

Reply:
status is used for compilation when HAVE_DECL_IBV_EXP_CQ_MODERATION macro is not defined

Reply:
fixed